manager/controlapi/service.go

```diff
@@ -424,6 +425,33 @@ func (s *Server) checkSecretExistence(tx store.Tx, spec *api.ServiceSpec) error
 	return nil
 }
 
+func doesServiceNeedIngress(srv *api.Service) bool {
+	if srv.Spec.Endpoint.Mode != api.ResolutionModeVirtualIP {
```

Comment:
`Endpoint` will be nil if unspecified in the API request, so be sure to check before dereferencing it.